packages/pytest-simcore/src/pytest_simcore/helpers/playwright.py

```diff
@@ -327,10 +327,18 @@ def expected_service_running(
     with log_context(logging.INFO, msg="Waiting for node to run") as ctx:
         waiter = SocketIONodeProgressCompleteWaiter(node_id=node_id, logger=ctx.logger)
         service_running = ServiceRunning(iframe_locator=None)
-        with websocket.expect_event("framereceived", waiter, timeout=timeout):
-            if press_start_button:
-                _trigger_service_start(page, node_id)
-            yield service_running
+
+        try:
+            with websocket.expect_event("framereceived", waiter, timeout=timeout):
+                if press_start_button:
+                    _trigger_service_start(page, node_id)
+        except TimeoutError:
```

Comment:
How do you guarantee that it was raised by  `websocket.expect_event` and not raised by `_trigger_service_start`
Moreover, that the cause of the timeout is the progress and not something else?

TIP:
- create `SocketIONodeProgressCompleteWaiter.last_progress` 
- check after the exception `if waiter.last_progress`
- otherwise let the exception raise